pkg/apis/autopilot/v1alpha1/autopilotrule.go

```diff
@@ -100,6 +99,50 @@ type AutopilotRuleSpec struct {
 	ActionsCoolDownPeriod int64 `json:"actionsCoolDownPeriod,omitempty"`
 }
 
+// AutopilotRuleStatus captures the status of various objects that are monitored as part of an autopilot rule
+type AutopilotRuleStatus struct {
+	// Objects is a map of objects that are monitored as part of a autopilot rule and their statuses
+	Objects map[RuleStatusObjectKey][]*AutopilotRuleObjectStatus `json:"objects"`
+}
+
+// AutopilotRuleObjectStatus represents status of a particular object that is being monitored by autopilot
+type AutopilotRuleObjectStatus struct {
+	// Name of the object
+	Name string `json:"name"`
+	// LastProcessTimestamp was the last time the object was processed
+	LastProcessTimestamp meta.Time `json:"lastProcessTimestamp"`
+	// State of the object
+	State RuleState `json:"state"`
+	// Message is the user friendly status
+	Message string `json:"message"`
+	// TODO add NextProcessTimestamp
+}
+
+// RuleState is the type for the state of a rule
+type RuleState string
+
+const (
+	// RuleStateInit is the initial state of the rule where monitorign has not yet begin
+	RuleStateInit RuleState = "Initializing"
+	// RuleStateNormal is when the rule is being monitored and is in normal state
+	RuleStateNormal RuleState = "Normal"
+	// RuleStateTriggered is when the rule has it's conditions met
+	RuleStateTriggered RuleState = "Triggered"
+	// RuleStateActivePending is when the rule has it's conditions met but the actions are
+	// not being performed yet.
+	RuleStateActivePending RuleState = "ActivePending"
+	// RuleStateActiveActionTaken is when the rule has it's actions already taken
+	// but still hasn't moved out of active status
+	RuleStateActiveActionTaken RuleState = "ActiveActionsTaken"
+	// RuleStateDeclined is when action was intentionally declined by autopilot
+	RuleStateDeclined RuleState = "Declined"
```

Comment:
"ActionDeclined" ? from the context it's not clear whether it's an action declined or rule is declined (from user's prospective. Let's assume they are not familiar with the system as good as we are )

Reply:
renamed to ActionsDeclined